WeatherApp.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { View, StyleSheet, ActivityIndicator } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import * as Location from 'expo-location';
+import useWeatherAPI from './useWeatherAPI';
+import CurrentWeatherScreen from './CurrentWeatherScreen';
+import ForecastScreen from './ForecastScreen';
+
+const Drawer = createDrawerNavigator();
+const API_KEY = 'd7e88c2243f84a3eb41183055231408';
+
+const WeatherApp = () => {
+  const navigation = useNavigation();
+  const [location, setLocation] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { status } = await Location.requestForegroundPermissionsAsync();
+        if (status === 'granted') {
+          const location = await Location.getCurrentPositionAsync({});
+          setLocation({
+            latitude: location.coords.latitude,
+            longitude: location.coords.longitude,
+          });
+        }
+      } catch (error) {
+        console.error('Error getting location:', error);
+      }
+    })();
+  }, []);
+
+  const { data: currentWeatherData, loading: currentWeatherLoading } = useWeatherAPI(
+    location ? `/current.json?q=${location.latitude},${location.longitude}` : '/current.json?q=Warwick,RI', // Default to Warwick if the location fails
+    API_KEY
+  );
+
+  const { data: forecastData, loading: forecastLoading } = useWeatherAPI(
+    location ? `/forecast.json?q=${location.latitude},${location.longitude}&days=7` : '/forecast.json?q=Warwick,RI&days=7',
```

Comment:
We're over fetching data here. 

If I only looked at the current weather, then we still ended up fetching 7 days of forecast information. In a small app this isn't a big deal, but when you work with larger data sets, you'll want to fetch exactly what you need and nothing more. 

As mobile devs, we need to transmit the lowest amount of data as possible. Users won't always have a fast connection.